queries.sql

```diff
@@ -1,11 +1,56 @@
-SELECT name FROM animals WHERE date_of_birth BETWEEN '2016-01-10' AND '2019-01-01' /*List the name of all animals born between 2016 and 2019.*/
-SELECT name FROM animals WHERE neutered='t' AND escape_attempts < 3; /*List the name of all animals that are neutered and have less than 3 escape attempts.*/
-SELECT date_of_birth from Animals WHERE name='Agumon' OR name ='Pikachu'; /* List date of birth of all animals named either "Agumon" or "Pikachu".*/
-SELECT name,escape_attempts FROM Animals WHERE weight_kg > 10.5; /* List name and escape attempts of animals that weigh more than 10.5kg */
-SELECT * FROM Animals WHERE neutered = 't'; /* Find all animals that are neutered.*/
-SELECT * FROM Animals WHERE  NOT name='Gabumon'; /* Find all animals not named Gabumon.*/
-SELECT * FROM Animals WHERE weight_kg >= 10.4 AND weight_kg <=17.3; /* Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)*/
+SELECT * FROM Animals WHERE name LIKE '%mon'; /*Find all Animals whose name ends in "mon".*/
+SELECT name FROM Animals WHERE date_of_birth BETWEEN '2016-01-10' AND '2019-01-01' /*List the name of all Animals born between 2016 and 2019.*/
+SELECT name FROM Animals WHERE neutered='t' AND escape_attempts < 3; /*List the name of all Animals that are neutered and have less than 3 escape attempts.*/
+SELECT date_of_birth from Animals WHERE name='Agumon' OR name ='Pikachu'; /* List date of birth of all Animals named either "Agumon" or "Pikachu".*/
+SELECT name,escape_attempts FROM Animals WHERE weight_kg > 10.5; /* List name and escape attempts of Animals that weigh more than 10.5kg */
+SELECT * FROM Animals WHERE neutered = 't'; /* Find all Animals that are neutered.*/
+SELECT * FROM Animals WHERE  NOT name='Gabumon'; /* Find all Animals not named Gabumon.*/
+SELECT * FROM Animals WHERE weight_kg >= 10.4 AND weight_kg <=17.3; /* Find all Animals with a weight between 10.4kg and 17.3kg (including the Animals with the weights that equals precisely 10.4kg or 17.3kg)*/
 
+
+/*
+Now, take a deep breath and... Inside a transaction delete all records in the Animals table, then roll back the transaction.
+*/
+BEGIN;
+TRUNCATE    Animals;
+DELETE FROM Animals;
+SELECT * FROM Animals; /* Check Whether table is now Empty*/
+ROLLBACK;
+
+/*
+    Inside a transaction:
+        Delete all Animals born after Jan 1st, 2022.
+        Create a savepoint for the transaction.
+        Update all Animals' weight to be their weight multiplied by -1.
+        Rollback to the savepoint
+        Update all Animals' weights that are negative to be their weight multiplied by -1.
+        Commit transaction
+*/
+
+BEGIN;
+DELETE FROM Animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT born_befor_2022;
+UPDATE Animals SET weight_kg = (weight_kg * -1);
+ROLLBACK TO born_befor_2022;
+SELECT  * FROM Animals; /* Verify data state */
```

Comment:
Between line 36 and 37 after you `ROLLBACK TO born_befor_2022;` there should be a query that `Updates all animals' weights that are negative to be their weight multiplied by -1.` as per project requirements.

![image](https://user-images.githubusercontent.com/38088818/171806802-63a57c05-c2f2-4830-9e24-16f07948123a.png)

For example:
```
    ...
    ROLLBACK TO SP1;
    UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
    COMMIT;
```